pkg/operator/operator_config.go

```diff
@@ -465,18 +464,20 @@ func (r *operatorConfigReconciler) makeAlertmanagerConfigs(ctx context.Context,
 		if ports := amSvc.Spec.Ports; len(ports) > 0 {
 			// Assume first port on service is the correct endpoint.
 			port := ports[0].Port
-			alertManagers = append(alertManagers, monitoringv1.AlertmanagerEndpoints{
-				Name:      amNamespacedName.Name,
-				Namespace: amNamespacedName.Namespace,
-				Port: intstr.IntOrString{
-					Type:   intstr.Int,
-					IntVal: port,
+			svcDNSName := fmt.Sprintf("%s.%s.svc.cluster.local:%d", amSvc.Name, amSvc.Namespace, port)
```

Comment:
It seems that `cluster.local` is adjustable by adjusting cluster-domain mentioned in [1]. It seems that GKE allows adjusting the cluster domain by customizing DNS [2].

Suggest using "%s.%s:%d" as it seems that it will resolve to the same address as long as a DNS service is set up [3]. If a DNS is not set up, I would assume "%s.%s.svc.cluster.local:%d" won't be usable either - hence the suggestion.

[1] https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/#introduction
[2] https://cloud.google.com/kubernetes-engine/docs/how-to/custom-kube-dns
[3] https://kubernetes.io/docs/concepts/services-networking/service/#dns

Reply:
Fantastic point. Thanks for catching this! I made the suggested change.